django/applications/catmaid/control/connector.py

```diff
@@ -553,9 +555,9 @@ def _connector_associated_edgetimes(connector_ids, project_id):
     FROM treenode_connector
     WHERE connector_id IN (%s)
       AND (relation_id = %s OR relation_id = %s)
-    ''' % (",".join(map(str, connector_ids), PRE, POST)))
+    ''' % (",".join(map(str, connector_ids), PRE, POST))) # FIXME: join doesn't take 3 arguments, format string consumes 3 args and this would return 1. Maybe bad parentheses?
```

Comment:
I was going to make that change, but I thought you might've had a reason to do it this way. I'll do the needed revision in this diff.

Reply:
I'm starting to adjust the diffs now; one thing I've seen (in other languages, so the limitation might not apply here) is that sometimes database engines don't have the ability to use SQL placeholders for things like table names (because placeholders are not meant to be for things that can change a query plan). I'm not sure if those limits might apply here. Hopefully travis will be able to test this adequately. (I'm also not familiar with that `ANY =` syntax you used)

Reply:
> sometimes database engines don't have the ability to use SQL placeholders for things like table names

Correct, because the query is prepared without the placeholders substituted (so they must only be values). I think the syntax should be `WHERE connector_id = ANY(%(connector_ids)s::bigint[])`. The `connector_ids` parameter is an array, so still a value to Postgres. It's a way to have arbitrary sized value inputs without needing to create query strings with arbitrary numbers of parameters.

Reply:
Oops, yes `WHERE connector_id = ANY(%(connector_ids)s::bigint[])` is what I meant, thanks Andrew!